### ISSUE TITLE
"AwaitSisterDeath"?

### ISSUE THREAD

**Issue by NikolasE** [ros/bond_core#84]
Maybe we should think about renaming these functions to something less dark... Isn't it also more of a TimeOut instead of certain death?

**Comment by mjcarroll** [comment]
I agree that the naming could be a little better, I don't have the bandwidth to tackle at the moment.  Any PRs in this direction would be appreciated.
